cmd/hlb/command/run.go

```diff
@@ -274,23 +273,42 @@ func Run(ctx context.Context, cln *client.Client, rc io.ReadCloser, info RunInfo
 	return err
 }
 
-func ModuleReadCloser(args []string) (io.ReadCloser, error) {
-	if len(args) == 0 {
-		return os.Open(DefaultHLBFilename)
-	} else if args[0] == "-" {
-		fi, err := os.Stdin.Stat()
+func ParseModuleURI(ctx context.Context, cln *client.Client, stdin io.Reader, uri string) (*ast.Module, error) {
+	if uri == "-" {
+		// Validate the stdin is piping into hlb.
```

Comment:
It feels like the stdin argument should have a stricter type, if there is no legitimate use case for passing in a generic `io.Reader`.

Reply:
Other than handling `os.Stdin` piping, `io.Reader` is sufficient so I didn't want to enforce the type across all library consumers. Maybe that is the right thing to do? How about checking if `stdin == os.Stdin`, does that work?

Reply:
Ended up just allowing `os.Stdin` from terminal as well as pipe:
```sh
❯ hlb run -
fs default() {
    image "alpine"
}
#1 resolve image config for docker.io/library/alpine:latest
#1 DONE 0.4s

#2 docker-image://docker.io/library/alpine:latest
#2 CACHED
```

Use `Ctrl+D` to terminate stdin input.

```sh
❯ echo 'fs default() { image "alpine" }' | hlb run -
#1 resolve image config for docker.io/library/alpine:latest
#1 DONE 0.4s

#2 docker-image://docker.io/library/alpine:latest
#2 CACHED
```